### PR TITLE
Allow `var` to have multiple declarations

### PR DESCRIPTION
**Pull request by imteekay**
- Add test to cover the new variable declaration list AST node
- Update all tests that have the `var` statement
- Create the `comma` token, parse the list AST node (with a list of declarations) — followed [this example](https://ts-ast-viewer.com/#code/G4QwTgBKYIwFwQM4BcwEsB2BzCBeCA5MgKYoEA0AUBDVOAEwIYCuAtgEbGT70DcQA)
- Type check each statement in the var declaration list

### CODE REVIEW

File: src/check.ts [hunk 1]
```diff
@@ -42,6 +43,9 @@ export function check(module: Module) {
         return t;
       case Node.TypeAlias:
         return checkType(statement.typename);
+      case Node.VariableDeclarationList:
+        statement.declarations.forEach(checkStatement);
+        return anyType;
```
Comment (imteekay) [inline src/check.ts:48]: Not sure which type I should return for `VariableDeclarationList` nodes.

File: README.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ npm run mtsc ./tests/singleVar.ts
 - [ ] Add let.
   - Then add use-before-declaration errors in the checker.
   - Finally, add an ES2015 -> ES5 transform that transforms `let` to `var`.
-- [ ] Allow var to have multiple declarations.
+- [x] Allow var to have multiple declarations.
   - You'll need to convert a Symbol's declaration into a list.
   - Check that all declarations have the same type.
```
Comment (imteekay) [inline README.md:47]: @sandersn, what did you mean by all declarations with the same type? When defining multiple variables with a `var`, we can have different types, right?

**Comment by imteekay** [comment]
cc @sandersn

File: README.md (hunk 2 above)
Comment (sandersn) [reply README.md:48]: whoops. I was unclear on this exercise. What I meant was that this is legal typescript:

```ts
var var1: string = 'test';
var var1: string = 'what, another one?'
var var1 = 'no type annotation needed'
```

But you cannot have a conflicting type or type annotation like `var var1 = 12`

This PR implements `var` statements that declare multiple symbols, not symbols with multiple declarations, each from different statements. I see now that both are valid interpretations of "Allow var to have multiple declarations".

File: src/types.ts [hunk 3]
```diff
@@ -72,13 +75,28 @@ export type Assignment = Location & {
   value: Expression;
 };
 
-export type Statement = ExpressionStatement | Var | TypeAlias | EmptyStatement;
+export type Statement =
+  | ExpressionStatement
+  | Var
```
Comment (sandersn) [inline src/types.ts:80]: `Var` isn't really a statement anymore, it's a child of VariableStatement.

File: src/check.ts (hunk 1 above)
Comment (sandersn) [reply src/check.ts:48]: That sounds about right for many nodes that don't have a sensible type. However, `Var` isn't really a statement anymore, so it should be handled in a different place from checkStatement. (this is true in the transformer and emitter too)

**Review by sandersn — COMMENTED** [review]
Looks about right. For long-term correctness, I would make `Var` not a statement, but it actually works fine for now.

File: README.md (hunk 2 above)
Comment (imteekay) [reply README.md:50]: Oooh, now I got it! Thanks for clarifying.
I created a new exercise "Allow var statements to declare multiple symbols" and this PR will handle that. So I can handle the "Allow var to have multiple declarations" in another PR.

File: src/check.ts (hunk 1 above)
Comment (imteekay) [reply src/check.ts:35]: That makes sense. I just made a change handle var as not a statement. https://github.com/imteekay/mini-typescript/pull/9/commits/147c648bd9ff4497b6f4db7db2ebcdf52e19b1ec

File: src/types.ts (hunk 3 above)
Comment (imteekay) [reply src/types.ts]: Just removed it https://github.com/imteekay/mini-typescript/pull/9/commits/147c648bd9ff4497b6f4db7db2ebcdf52e19b1ec

**Pull request merged**